fsharp-backend/tests/FuzzTests/ExecutePureFunctions.FuzzTests.fs

```diff
@@ -128,17 +128,20 @@ module Generators =
               (fun i (v : RT.DType) -> (id i, $"{v.toOldString().ToLower()}_{i}"))
               paramTypes
 
-          // FSTODO: occasionally use the wrong return type
-          // FSTODO: can we use the argument to get this type?
+          let! returnType =
+            Gen.frequency [ (98, Gen.constant returnType)
+                            (2, G.RuntimeTypes.dType) ]
```

Comment:
I believe this matches your expectation for the FSTODO? In any case, I've tested that this works OK.

Reply:
Yup, perfect!